### ISSUE TITLE
Static method arrow function does not name anonymous function

### ISSUE THREAD

**Issue by reggi** [babel/babel#7425]
I'd expect `M.blue.name` to be `blue`, but for some reason arrow functions don't have a name. Is this a bug or intentional?

```
class M {
static green () { return 'green' }
static blue = () => 'green'
}
module.exports = M
```

```
var M = function () {
  function M() {
    _classCallCheck(this, M);
  }

  _createClass(M, null, [{
    key: 'green',
    value: function green() {
      return 'green';
    }
  }]);

  return M;
}();

M.blue = function () {
    'green'
};
```

Guess I'd love it to be like this instead:

```
M.blue = function blue () {
    'green'
};
```

**Comment by thymikee** [comment]
Can you provide a repro? What are the plugins/presets you're using?
I can only verify that with
```
"plugins": [
  "@babel/plugin-proposal-class-properties"
]
```
produces this output
```js
class M {
  static green() {
    return 'green';
  }

}

Object.defineProperty(M, "blue", {
  configurable: true,
  enumerable: true,
  writable: true,
  value: () => 'green'
});
module.exports = M;
```
and `M.blue.name === 'value'`, so it looks like a bug.

**Comment by Andarist** [comment]
@thymikee I guess he is using `loose` mode which doesn't use `defineProperty` for static methods, just a simple assignments.

Also `defineProperty` usage is related to what you can see - `value` inferred as name of this function because it's used as value (:<) of the descriptor's property called `value`

**Comment by thymikee** [comment]
Oh, that's unfortunate. Looks like that arrow should be eventually replaced with a named function in both cases, right?

**Comment by Andarist** [comment]
Ain't sure if class properties (static or not) have inferred names - I **think** spec says they can, but I'm not the best ecma spec reader :/

**Comment by thymikee** [comment]
I'm not an expert either, but looking at https://tc39.github.io/proposal-class-public-fields/#initialize-public-instance-fields I could tell the spec says `.name` should be derived from the field name:

<img width="535" alt="screen shot 2018-02-27 at 23 30 08" src="https://user-images.githubusercontent.com/5106466/36758967-3044b55a-1c16-11e8-88a7-2112db41f059.png">

**Comment by Andarist** [comment]
Yeah, I suspect this is right reasoning. I guess the only way to 'fix' it is to transform directly to named functions, because at later point any other transform won't have enough info to infer the name correctly (just like in the `value` example).

**Comment by vivek12345** [comment]
Does this still need help? Can I pick this up?
